### ISSUE TITLE
Reshade for PotPlayer - advanced postprocessing filter (deband for example)

### ISSUE THREAD

**Issue by DanillaTech** [23rd/PotPlayerExtentions#5]
Hey. May I ask you to make reshade (https://github.com/crosire/reshade) available for PotPlayer? All it need for extension just call  "HostLoadLibrary("reshade.dll")"?

I use it for debanding, it works better and more stable than MadVR

You need to extract reshade's dll and put it to "lib" folder. You can try reshade works with dll injectors
There is comparing
Original (see in sky gradient) https://d.radikal.ru/d33/1901/db/cbcf3210d257.png
MadVR https://c.radikal.ru/c13/1901/21/7b18a56c5869.png
Reshade https://c.radikal.ru/c34/1901/89/4faeb772898a.png

**Comment by 23rd** [comment]
Hi.
I don't see much difference between the original and reshade screenshots. Except that reshade's picture is slightly lighter.
Are you sure the extension really needs to have this dll?

**Comment by DanillaTech** [comment]
> Hi.
> I don't see much difference between the original and reshade screenshots. Except that reshade's picture is slightly lighter.
> Are you sure the extension really needs to have this dll?

Yeah, i think. MadVR is quite popular render for 'best' quality), but in our time most popular of it's feature is deband. I don't know what exactly it does for 'best' quality, but my Intel HD4000+i73632QM lags on deband+60FPS SVP . From another side reshade's deband filter can work better and more customizible and it takes less GPU.
If you do not see difference, it's contrasted comparison.
![image](https://user-images.githubusercontent.com/8999560/51540137-bdb89980-1e66-11e9-993d-25099158d289.png)


//But from another side actually all what I need from reshade is deband filter. It can be ported for HLSL what PotPlayer support, but I can not do it. I  know that potplayer support dll's loading - it what torrent extension does. I modified torrent extension for loading reshade's dll . Reshade works in my case **only on Direct3d11 render**, although reshade for games works on OGL, D3D9

[Extention.zip](https://github.com/23rd/PotPlayerExtentions/files/2783059/Extention.zip)

**Comment by 23rd** [comment]
> Reshade works in my case only on Direct3d11 render, although reshade for games works on OGL, D3D9

So if I will insert the reshade dll, everyone who use this extension will see anything only on Direct3D11?

**Comment by DanillaTech** [comment]
>So if I will insert the reshade dll, everyone who use this extension will see anything only on Direct3D11?
May be) Reshade does not presented as program for players, but in forum you can see it works https://reshade.me/forum/general-discussion/2672-is-it-possible-to-use-reshade-on-a-video-player#19267

**Comment by DanillaTech** [comment]
А, автор русский)

В принципе удалось 'удобно' запилить решейд, для его запуска в сборке использую костыль, включив его в плагин торрентокачалки. Уровень дебандинга можно установить как в 100 раз выше мадвра до превращения картинки в мыло, можно выбрать среднее мыльцо и очень низкое-высокоточное

![image](https://user-images.githubusercontent.com/8999560/52527423-77be5b00-2cd9-11e9-96c7-5286acb2918d.png)

Но всё-таки интереснее решение с минимальными костылями, типа портированных шейдеров/быстрого и корректного подключения решейда и его работы - сейчас при переходе между файлами требуется останавливать и запускать воспроизведение.

Если что, сборка с решейдом https://rutracker.org/forum/viewtopic.php?t=5687393

**Comment by BinToss** [comment]
Wait a minute! This issue is outside the scope of TwitchPotPlayer!
This TwitchPotPlayer is an extension for PotPlayer written in AngelScript. Its purpose is to integrate Twitch.tv similarly to the built-in YouTube extension for PotPlayer. They provide media parsing and URL interactions and nothing more.

As much as I like the proposal of 'integrating' the ReShade framework into PotPlayer, this suggestion is irrelevant to TwitchPotPlayer and likely wouldn't be possible as an AngelScript project.
